be/src/exec/spill/spill_components.cpp

```diff
@@ -72,52 +72,24 @@ Status RawSpillerWriter::yieldable_flush_task(workgroup::YieldContext& yield_ctx
     if (state->is_cancelled()) {
         return Status::OK();
     }
-    const auto& serde = _spiller->serde();
 
-    auto block_st = get_block_from_ctx([&]() -> StatusOr<BlockPtr> {
-        if (!yield_ctx.task_context_data.has_value()) {
-            spill::AcquireBlockOptions opts{.query_id = state->query_id(),
-                                            .plan_node_id = options().plan_node_id,
-                                            .name = options().name,
-                                            .direct_io = state->spill_enable_direct_io()};
-            ASSIGN_OR_RETURN(auto block, _spiller->block_manager()->acquire_block(opts));
-            COUNTER_UPDATE(_spiller->metrics().block_count, 1);
-            yield_ctx.task_context_data = FlushContext{block};
-        }
-        auto flush_ctx = std::any_cast<FlushContext>(yield_ctx.task_context_data);
-        return flush_ctx.block;
-    });
-    ASSIGN_OR_RETURN(auto block, std::move(block_st));
+    DCHECK(yield_ctx.task_context_data.has_value()) << "task context must be set before invoke flush task";
+    auto block = std::any_cast<FlushContextPtr>(yield_ctx.task_context_data)->block;
 
-    // TODO: reuse io context
-    SerdeContext spill_ctx;
-    int64_t time_spent_ns = 0;
     yield_ctx.total_yield_point_cnt = 1;
-    {
-        TRY_CATCH_ALLOC_SCOPE_START()
-        // flush all pending result to spilled files
-        size_t num_rows_flushed = 0;
-        RETURN_IF_ERROR_EXCEPT_YIELD(mem_table->flush([&](const auto& chunk) {
-            SCOPED_RAW_TIMER(&time_spent_ns);
-            num_rows_flushed += chunk->num_rows();
-            RETURN_IF_ERROR(serde->serialize_to_block(spill_ctx, chunk, block));
-            RETURN_IF_NEED_YIELD(yield_ctx.wg, yield, time_spent_ns);
-            return Status::OK();
-        }));
-        TRACE_SPILL_LOG << "spill flush rows:" << num_rows_flushed << ",spiller:" << this;
-        RETURN_IF_YIELD(yield);
-        TRY_CATCH_ALLOC_SCOPE_END();
-    }
 
     // be careful close method return a not ok status
     // then release the pending memory
+    ASSIGN_OR_RETURN(auto data, mem_table->get_serialized_data());
```

Comment:
It seems yield won't work?

Reply:
> It seems yield won't work?

here we  only get the result, the serialization process has already been done in the computing thread, no yield is needed here.